guides/source/working_with_javascript_in_rails.md

```diff
@@ -3,561 +3,197 @@
-  });
-```
-
-This code fetches data from "/test", and then appends the result to the element
-with an id of `results`.
+* How to use Rails without the need for a Node, Yarn, or a JavaScript bundler
+* How to create a new Rails application using esbuild, rollup, or webpack to bundle your JavaScript
+* What Turbo is
+* How to use the Turbo HTML helpers provided by Rails
 
-Rails provides quite a bit of built-in support for building web pages with this
-technique. You rarely have to write this code yourself. The rest of this guide
-will show you how Rails can help you write websites in this way, but it's
-all built on top of this fairly simple technique.
+———————————————————————————————————————
 
-Unobtrusive JavaScript
-----------------------
+Import maps
+————————————
 
-Rails uses a technique called "Unobtrusive JavaScript" to handle attaching
-JavaScript to the DOM. This is generally considered to be a best-practice
-within the frontend community, but you may occasionally read tutorials that
-demonstrate other ways.
+[Import maps](https://github.com/rails/importmap-rails) let you import JavaScript modules using logical names that map to versioned/digested files directly from the browser. Import maps are the new default for Rails 7, allowing anyone to build modern JavaScript applications using JavaScript libraries built for ESM without the need for transpiling or bundling.
 
-Here's the simplest way to write JavaScript. You may see it referred to as
-'inline JavaScript':
+Applications using import maps do not need Node or Yarn to function. If you plan to use Rails with importmap-rails to manage your JavaScript dependencies, you can use Rails without needing to install Node or Yarn.
 
-```html
-<a href="#" onclick="this.style.backgroundColor='#990000';event.preventDefault();">Paint it red</a>
-```
+When using import maps, no separate build step is required, just start your server with `rails s` and you are good to go.
 
-When clicked, the link background will become red. Here's the problem: what
-happens when we have lots of JavaScript we want to execute on a click?
-
-```html
-<a href="#" onclick="this.style.backgroundColor='#009900';this.style.color='#FFFFFF';event.preventDefault();">Paint it green</a>
-```
+### Adding NPM packages with importmap-rails
+To add new packages to your import map-powered application, run the `importmap pin` command from your terminal:
 
-Awkward, right? We could pull the function definition out of the click handler,
-and turn it into a function:
-
-```js
-window.paintIt = function(event, backgroundColor, textColor) {
-  event.preventDefault();
-  event.target.style.backgroundColor = backgroundColor;
-  if (textColor) {
-    event.target.style.color = textColor;
-  }
-}
+```bash
+./bin/importmap pin alpinejs
 ```
 
-And then on our page:
-
-```html
-<a href="#" onclick="paintIt(event, '#990000')">Paint it red</a>
+Then, import the package into `application.js` as usual:
+```javascript
+import Alpine from 'alpinejs'
```

Comment:
No personal opinion on alpine, but I think it would be better to use Stimulus as an example since it's in the Rails tent.

Reply:
That's fair — I didn't use Stimulus/Turbo since those are going to be installed automatically for most folks who are using import maps, but maybe that's over thinking it a bit. I'll make that switch.

Reply:
Hmmm, that's a good point.

Maybe use the react example from https://github.com/rails/importmap-rails#how-do-importmaps-work so it's consistent?